lib/cook_book.rb

```diff
@@ -0,0 +1,27 @@
+class CookBook
+  attr_reader :recipes
+
+  def initialize
+    @recipes = []
+  end
+
+  def add_recipe(recipe)
+    @recipes << recipe
+  end
+
+  def ingredients
+    all_ingredients = []
+    @recipes.each do |recipe|
+      recipe.ingredients_required.each do |ingredient_required, quantity|
+        all_ingredients << ingredient_required.name 
+      end
+    end
+    all_ingredients.uniq
```

Comment:
Great job here. I wonder if you could figure out a way to do it with another enumerable other than .each :)